File: src/main/lua/external.lua
require "lua-channel"

function producer(char, ch)
    PAR(
        function()
            while true do
                KEYSTATE( ch, char )
                print("value send " + char)
            end
        end
    )
end

function consumer(ch)
    PAR(
        function()
            while true do
                local char = ch:IN()
                print("Value get :" + char)
            end
        end
    )
end

function main()
    local ch = Channel:new()
    producer("a", ch)
    consumer(ch)
end

File: OCCAM/src/main/python/external.py
#!/usr/bin/env python

from csp.csp import *

@process
def producer(channel, n):
     channel.write(n)
     print "send value : " + n
     channel.poison()
     return

@process
def consumer(channel):
     print "get value : " + channel.read()

chan = Channel()
Par(consumer(chan), producer(chan, 'a')).start()